### PR TITLE
Refactor to class

### PR DESCRIPTION
**Pull request by librasteve**
hi @FCO 

I have merged your latest v0.0.1 Cromponent into my examples and applied the changes I mention in Issue #5 to refactor the Cromponent code to a "re-entrant" class

As you can see there is minimal change to the /bin scripts and only really a class wrapper around Cromponents.rakumod.

Hopefully I have convinced you with the case I made in the issue thread and the example code in Cro-Website-Basic examples...?

Happy to answer any questions on this.

### CODE REVIEW

**Comment by librasteve** [comment]
this PR replaces the previous ones

File: bin/cromponent.raku [hunk 1]
```diff
@@ -54,7 +55,7 @@ class Todo {
 my $routes = route {
 	my @todos = do for <blablabla blebleble> -> $data { Todo.new: :$data }
 	get  -> {
-		template-with-components Q:to/END/, { :@todos };
+		template-with-components $cromponent, Q:to/END/, { :@todos };
```
Comment (FCO) [inline bin/cromponent.raku:58]: If going to do that, why not calling it as a method of $cromponets?

**Review by FCO — COMMENTED** [review]
@librasteve thank you for your PR. I'm still grasping to understand why would you like to add the same components (or components with the same name) multiple times. I'm currently on vacations with few time with access to my computer (my wife tries to not allow me to use the computer while I'm on vacation). But I'm trying to find a way to stick with not needing to have cromponents objects I feel like it will be a problem when distributing routes and I want to use component defined in different places in the same "list of components"... I think having a trait "is cromponent" would help on that.

**Comment by FCO** [comment]
I've been thinking... we may be able to inject the components inside Cro's dynamic RouteSet... that way it will not be in conflict on your example, and it would, maybe, not need the location addition, because it could get it from the RouteSet itself... I'll need some time to test it...

**Comment by FCO** [comment]
@librasteve I just did what I said. there is a new branch (spread-routes (https://github.com/FCO/Cromponent/tree/spread-routes)) where I store the components inside the RouteSet. Could you take a look if that fixes the problem to you? We have an example [here](https://github.com/FCO/Cromponent/blob/spread-routes/bin/spreaded-routes.raku) and when running that, you can test with:
- https://0.0.0.0:3000/cromponent
- https://0.0.0.0:3000/macro
- https://0.0.0.0:3000/table-nest
- https://0.0.0.0:3000/table-nest-func
- https://0.0.0.0:3000/table-nest-lib

here it seems to be working. It would be great if you could test and let me know your thoughts.

Thanks.

**Comment by librasteve** [comment]
@FCO - this looks very good - it resolves the issue #5 I raised - thank you

PS. I was not able to understand how to run the https://github.com/FCO/Cromponent/blob/spread-routes/bin/spreaded-routes.raku file ... but I can see that you have a site with multiple included routes and that was the issue

**Pull request closed without merging**